search_knn.go

```diff
@@ -225,7 +225,7 @@ func adjustRequestSizeForKNN(req *SearchRequest, numIndexPartitions int) int {
 
 // heap impl
 type scoreHeap struct {
-	scoreBreakdown []*[]float64
+	scoreBreakdown []*map[int]float64
```

Comment:
Does this need to be an array of maps?

Also the map does not need to be a pointer, just `[]map[int]float64` should do.